src/merge_maps_tool.cpp

```diff
@@ -179,6 +198,119 @@ bool MergeMapTool::MergeMapCallback(slam_toolbox::MergeMaps::Request &req,
   // against them to make a new inter-graph contraint list to optimize over
   // then generate new composite map
   // (this same technique can then be applied with manual loop closures)
+    int id = 0;
+
+    karto::Pose2 corrected_pose;
+    karto::LocalizedRangeScan* pScan;
+    karto::LocalizedRangeScanVector transformed_scans;
+    std::vector<karto::LocalizedRangeScanVector> vector_of_scans = scans_vec;
+    karto::LocalizedRangeScan* pScan_copy;
+
+    for(std::vector<karto::LocalizedRangeScanVector>::iterator it_LRV = vector_of_scans.begin(); it_LRV!= vector_of_scans.end(); it_LRV++)
+    {
+      id++;
+      for ( karto::LocalizedRangeScanVector::iterator iter = (*it_LRV).begin(); iter != (*it_LRV).end();iter++ )
+      {
+         pScan= *iter;
+         pScan_copy = pScan;
+
+//        tf::Transform submap_correction = submap_init[id-1].inverse()*tf_map[id];
+        tf::Transform submap_correction = tf_map[id];
+
+        //TRANSFORM BARYCENTERR POSE
+        karto::Pose2 baryCenter_pose = pScan_copy->GetBarycenterPose();
+        tf::Transform baryCenter_pose_tf;
+        baryCenter_pose_tf.setOrigin(tf::Vector3(baryCenter_pose.GetX(), \
+                                    baryCenter_pose.GetY(), 0.));
+        baryCenter_pose_tf.setRotation(tf::createQuaternionFromRPY(0, 0, baryCenter_pose.GetHeading()));
+
+        tf::Transform baryCenterPose_corr;
+        baryCenterPose_corr = submap_correction*baryCenter_pose_tf;//*
+
+        karto::Pose2 karto_baryCenterPose_corr = \
```

Comment:
`const karto::Pose2 karto_baryCenterPose_corr(args)` use the constructor, you're incurring an extra copy. Please do for all instances below